src/shared/components/exonsCharts/ExonsChartStore.ts

```diff
@@ -0,0 +1,252 @@
+                    t: EnsemblTranscriptExt,
+                    prevLongest: StructuralVariantExt | undefined): boolean {
+        let longerThanPreviousLongest = true;
+        fusion.totalWidth = fusion.totalWidth ? fusion.totalWidth : 0;
+        if (prevLongest) {
+            prevLongest.totalWidth = prevLongest.totalWidth ? prevLongest.totalWidth : 0;
+            longerThanPreviousLongest = prevLongest ? fusion.totalWidth > prevLongest.totalWidth : true;
+        }
+        return (fusion.totalWidth > t.totalWidth) && !fusion.isLeftAligned && longerThanPreviousLongest;
+    }
+
+    getFusionDetails(t: EnsemblTranscriptExt): StructuralVariantExt[] {
+        let prevFusions: StructuralVariantExt[] = [];
+        let prevSite1LongestVal: number = 0;
+        return this.computedFusions
+            // get only fusion data for the given transcript id
+            .filter(fusion => {
+                return (
+                    fusion.site1EnsemblTranscriptId === t.transcriptId ||
+                    fusion.site2EnsemblTranscriptId === t.transcriptId
+                );
+            })
+            .map(fusion => {
+
+                const prevSite1Longest = this.getSite1LongestFusion(prevFusions);
+                const isCurrentFusionHasLongestSite1 = this.hasLongestSite1(fusion, t, prevSite1Longest);
+
+                prevSite1LongestVal = 0; // re-initialize prev site1 longest val
+
+                // initialize total width
+                fusion.totalWidth = fusion.totalWidth ? fusion.totalWidth : 0;
+
+                // initialize delta-x for both ref gene and fusion
+                t.deltaX = isCurrentFusionHasLongestSite1 ? fusion.totalWidth - t.totalWidth : t.deltaX;
+                fusion.deltaX = isCurrentFusionHasLongestSite1 ? 0 : t.deltaX;
+
+                // when current fusion has longest site1
+                if (isCurrentFusionHasLongestSite1) {
+                    prevFusions = this.updateDeltaX(prevFusions, t);
+                } else {
+                    // get previously longest site1 total width
+                    if (prevSite1Longest) {
+                        prevSite1LongestVal =
+                            prevSite1Longest.totalWidth ? prevSite1Longest.totalWidth : prevSite1LongestVal;
+                    }
+                    // determine the delta-x
+                    if (!fusion.isLeftAligned && (fusion.totalWidth > t.totalWidth) && (prevSite1LongestVal > 0)) {
+                        fusion.deltaX = prevSite1LongestVal - fusion.totalWidth;
+                    } else {
+                        fusion.deltaX = t.deltaX;
+                    }
+                }
+                prevFusions.push(fusion);
+                return fusion;
+            });
+    }
+
+    /**
+     * Associate fusions with their reference gene transcipts
+     * @returns {EnsemblTranscriptExt[]}
+     */
+    @computed
+    get fusionsByReferences(): EnsemblTranscriptExt[] {
```

Comment:
this name doesn't make any sense. this also changes referenceTranscripts in place